src/socket/reader.ts

```diff
@@ -53,7 +53,8 @@ export class WebSocketMessageReader extends AbstractStreamMessageReader {
         if (this.state === 'initial') {
             this.events.splice(0, 0, { message });
         } else if (this.state === 'listening') {
-            super.readMessage(message, this.callback!);
+            const data = JSON.parse(message);
+            this.callback!(data);
```

Comment:
could it happen that `this.callback` is assigned to late if `readMessage` is called via L40?

Reply:
good catch, I will move assigning `callback` before processing buffered messages

Reply:
done